packages/node_modules/@webex/plugin-meetings/src/meeting/index.js

```diff
@@ -183,54 +256,58 @@ export default class Meeting {
-      sendAudio: true,
-      sendVideo: true,
-      receiveAudio: true,
-      receiveVideo: true,
-      pstn: false,
-      sendShare: false,
-      receiveShare: false
-    };
-    return this.joinMeeting()
-      .then(() =>
-        this.getLocalMedia({
+    const {mediaDirection} = options;
+    this.setCorrelationId(this.id);
+    this.state.join();
+    return MeetingUtil.joinMeeting(this)
+      .then((response) => {
+        this.setLocusJoin(response);
+        this.state.local();
+        return Media.getLocalMedia({
           sendAudio: mediaDirection.sendAudio,
           sendVideo: mediaDirection.sendVideo,
           sendShare: mediaDirection.sendShare
-        }))
-      .then((streams) => this.addMedia(streams))
+        });
+      })
+      .catch((err) => {
+        console.log('Meeting:index#join --> Error joining the call on getLocalMedia, ', err);
+        throw err;
+      })
+      .then((streams) => {
+        this.state.connect();
+        this.setLocalVideo(streams);
+        return Media.addMedia(mediaDirection, PeerConnectionManager, streams);
+      })
+      .catch((err) => {
+        console.log('Meeting:index#join --> Error joining the call on addMedia, ', err);
+        throw err;
+      })
       .then(([mediaPeerConnection, screenPeerConnection]) => {
-        if (screenPeerConnection && screenPeerConnection.sdp) {
-          screenPeerConnection.sdp += `${SDP.A_CONTENT_SLIDES}${SDP.CARRIAGE_RETURN}`;
-        }
-
-        // Cache these locally for later use
-        this.mediaPeerConnection = mediaPeerConnection;
-        this.screenPeerConnection = screenPeerConnection;
-
-        // Receiving remote media streams
-        this.mediaPeerConnection.ontrack = (event) => {
-          [this.mediaStream, this.localMediaStream] = event.streams;
-          document.getElementById('remotevideo').srcObject = this.mediaStream;
-        };
-
-        // Receive remote screen streams TODO
-        this.screenPeerConnection.ontrack = (event) => {
-          [this.screenStream] = event.streams;
-          document.getElementById('screenvideo').srcObject = this.screenStream;
-        };
-
+        screenPeerConnection = PeerConnectionManager.setContentSlides(screenPeerConnection);
+        this.setPeerConnections(mediaPeerConnection, screenPeerConnection);
+        this.setRemoteStreams(this.mediaPeerConnection);
+        this.setScreenStreams(this.screenPeerConnection);
         return RoapAdapter.initialize();
       })
-      .then(() =>
+      .catch((err) => {
+        console.log('Meeting:index#join --> Error joining the call on roap initialization, ', err);
+        throw err;
+      })
+      .then(() => {
         RoapInterface.sendRoapMediaRequest({
           sdps: [this.mediaPeerConnection.sdp, this.screenPeerConnection.sdp],
           roapSeq: this.roapSeq,
           meeting: this // or can pass meeting ID
-        }));
+        })
+          .then(() => {
+            this.state.establish();
```

Comment:
Do these state changes have side effects (calling other methods) or just updating the "state" prop?